cmd/syncthing/main.go

```diff
@@ -931,11 +934,27 @@ func copyFile(src, dst string) error {
 	return nil
 }
 
-func startAuditing(mainService *suture.Supervisor) {
-	auditFile := timestampedLoc(locAuditLog)
-	fd, err := os.OpenFile(auditFile, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0600)
-	if err != nil {
-		l.Fatalln("Audit:", err)
+func startAuditing(mainService *suture.Supervisor, auditFile string) {
+
+	var fd io.Writer
+	var err error
+	var auditDest string
+
+	if auditFile == "-" {
+		fd = os.Stdout
+		auditDest = "stdout"
+	} else if auditFile == "--" {
+		fd = os.Stderr
+		auditDest = "stderr"
+	} else {
+		if auditFile == "" {
+			auditFile = timestampedLoc(locAuditLog)
+		}
+		fd, err = os.OpenFile(auditFile, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0600)
```

Comment:
Yes.  That had been my intention at first since that was the previous behaviour when writing an audit file.  Also, when the program resets so do the event id numbers, so I wasn't sure that it was beneficial to append the event data.  However there are still time stamps, so I will make it so that if an explicitly set audit file exits it will append, but an automatically generated one will still trigger an error.

By the way, while testing the new code, I found that -logfile=<name> doesn't work.  Logs always go to stdout.